docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,177 @@
+
+## Overview
+
+Data tend to be created in logical batches by machines, people, and
+pipelines. Detecting these logical events from Amazon S3 events alone is
+complex and requires extensive custom logic.
+
+Quilt's *Event-Driven Packaging* service (EDP) smartly groups one
+or more Amazon S3 object events into a single event
+in [AWS EventBridge](https://aws.amazon.com/eventbridge/) so that
+you can easily trigger logical events like package creation that
+depend on batches rather than on individual files.
+
+Quilt EDP works well with services like [AWS Storage
+Gateway](https://aws.amazon.com/storagegateway/) and [AWS Data
+Sync](https://aws.amazon.com/datasync/) that write objects to Amazon S3.
+
+> Any AWS service or action that generates S3 object events may trigger the EDP service.
+
+## Requirements
+
+1. A pre-existing VPC that includes a [NAT
+Gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html)
+or the following [VPC
+endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html#concepts-vpc-endpoints):
+  - Amazon S3
+  ([gateway](https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html)
+  or
+  [interface](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html))
+  - EventBridge (interface)
+2. [Enable EventBridge S3
+Events](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications-eventbridge.html)
+for all S3 buckets to be monitored by EDP
+
+![](../imgs/edp-event-bridge-s3-properties.png)
+
+## Deployment
+
+To use EDP in your Quilt instance, you will use a standalone
+CloudFormation template to configure when, and how, events get
+generated.
+
+### Parameters
+
+The following parameters need to be defined in the CloudFormation
+template during deployment:
+
+- `VPC`: Same as the existing Quilt instance.
+- `SecurityGroup`: Same as the existing Quilt instance.
+- `Subnets`: Same two subnets as the existing Quilt instance.
+- `BucketName`: Name of the Amazon S3 bucket to monitor.
+- `BucketIgnorePrefixes`: List of bucket path segments to ignore
+- `BucketPrefixDepth`: The number of `/`-separated *common* path segments
+at the beginning of an S3 object key. Default value is `2`.
+- `BucketThresholdDuration`: Trigger a notification when this number
+of seconds has elapsed since the last object event in the S3 bucket 
+occurred. Default value is `300` seconds.
+- `BucketThresholdEventCount`: Trigger a notification when this
+number of files have been created (since the prior trigger)
+- `DBUser`: Username for EDP RDS instance
+- `DBPassword`: Password for EDP RDS instance
+- `EventBusName`: Name of custom EventBridge event bus that recieves events
+
+![](../imgs/edp-cloudformation-parameters.png)
+
+## How EDP works
+
+1. EDP monitors S3 object events for _s3://source-bucket_
+2. After a fixed number of events (`BucketThresholdEventCount`) or
+a maximum duration (`BucketThresholdDuration`) EDP creates a
```

Comment:
```suggestion
a maximum duration within a common prefix (`BucketThresholdDuration`) EDP creates a
```
suggest adding "within a common prefix" to reduce ambiguity (per earlier conversation)